backend/services/vector_store.py

```
@@ -526,14 +516,95 @@ async def delete_document(self, document_id: str, collection_name: str = "defaul
+            # Get count before deletion for logging
+            count_before = query.count()
+            
+            if count_before == 0:
+                return 0
+            
+            
+            # Delete the matching documents
+            deleted_count = query.delete(synchronize_session=False)
+            db.commit()
+            
+            return deleted_count
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)